ChallengesWithTestsMarkVII/ChallengesSet06.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ChallengesWithTestsMarkVII
+{
+    public class ChallengesSet06
+    {
+        public bool CollectionContainsWord(IEnumerable<string> words, string word, bool ignoreCase)
+        {
+
+            bool doesContain = false;
+            if (words == null)
+                return false;
+
+            if (ignoreCase)
+            {
+                foreach (var wrd in words)
+                {
+                    if (wrd != null && wrd.ToLower() == word.ToLower())
+                    {
+                        doesContain = true;
```

Comment:
If this is reached, then the method's final result is known; no need to keep checking words, go ahead and `return true;`